test/e2e/apimachinery/watchlist.go

```diff
@@ -39,13 +40,7 @@ import (
 var _ = SIGDescribe("API Streaming (aka. WatchList)", framework.WithSerial(), feature.WatchList, func() {
 	f := framework.NewDefaultFramework("watchlist")
 	ginkgo.It("should be requested when WatchListClient is enabled", func(ctx context.Context) {
-		// TODO(p0lyn0mial): use https://github.com/kubernetes/kubernetes/pull/123974
-		//  instead of using directly clientfeatures.ReplaceFeatureGates
-		prevClientFeatureGates := clientfeatures.FeatureGates()
-		defer func() {
-			clientfeatures.ReplaceFeatureGates(prevClientFeatureGates)
-		}()
-		clientfeatures.ReplaceFeatureGates(newEnabledWatchListClientFeatureGateRegistry(prevClientFeatureGates))
+		clientfeaturestesting.SetFeatureDuringTest(ginkgo.GinkgoTB(), clientfeatures.WatchListClient, true)
```

Comment:
this one could actually use the component-base like this, right?

https://github.com/kubernetes/kubernetes/blob/master/test/integration/storageversionmigrator/storageversionmigrator_test.go#L50

Reply:
It turns out we have to set this feature via component-base because the gates were already replaced. 

Previously `pull-kubernetes-e2e-gce-cos-alpha-features` failed on: `clientfeatures.FeatureGates(): *features.clientAdapter does not implement featureGatesSetter interface`